### ISSUE TITLE
Failed Task cannot be resumed

### ISSUE THREAD

**Issue by Joker12abuqasf** [781flyingdutchman/background_downloader#102]
Hi, mr developer, I used the background downloader library, which is very, very wonderful, but I could not resume the failed task, and I did not find evidence that explains whether this is possible or not. If it can resume the failed task, please explain this through the code. Thank you.

**Comment by 781flyingdutchman** [comment]
Hi, you cannot resume a failed task. Only paused tasks can be resumed (and you, the developer, would pause the task). If a task fails you'll get some information in the TaskException that is included in the TaskStatusUpdate, and based on that you can decide what to do next. 
If you want to try the same download again, then create a new task and download or enqueue it. You can also let the downloader retry failed tasks by setting the task's `retries` field to something like 5, before starting the download.

**Comment by Joker12abuqasf** [comment]
Thank you for that but I used that ok can i pause the task when the internet suddenly shuts down (wifi) instead of it failing thank you

**Comment by 781flyingdutchman** [comment]
An Internet connection problem while a download is already in progress will eventually lead to a task failure, not a task pause. If you can detect the Internet going down, eg using another Flutter package, then you could possibly trigger a pause before the task fails.

**Comment by 781flyingdutchman** [comment]
Implemented in V7.9.1:
Failed download tasks can now be resumed (under certain conditions) even if the `Task.allowPause` field is false. Resuming a failed task will attempt to continue the download where the failure occurred.  If `retries` are set to a value >0 then retries will also first attempt to resume, and only start from scratch if that fails.

Tasks can only resume if the ETag header provided by the server is strong, and equal to the ETag at the moment the download was paused/failed, or if it is not provided at all.
